src/app/utilities/direction.utility.ts

```diff
@@ -9,7 +12,8 @@ export function getDirection(key: string): Direction {
     case 'ArrowDown':
       return Direction.DOWN;
     case 'ArrowRight':
-    default:
       return Direction.RIGHT;
+    default:
+      return currentDirection;
```

Comment:
Hmm, I'll propose a different approach here: if the pressed button is not one of the 4 arrows, we could return null. Then, we can see that the new direction is null so we won't proceed with the change of the snake's direction